server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -70,6 +73,19 @@ public void release(LockInfo lockInfo) throws IOException {
         }
     }
 
+    public String fetchLock(String filenamePrefix, String acquirerId) throws IOException {
+        Collection<String> lockFiles = lockDirectory.listFilesByPrefix(filenamePrefix);
```

Comment:
Will this perform a paginated call?

Reply:
Yes

Reply:
But for a given shard, primary term and generation, there can't be more than one snapshot holding the lock.

Reply:
There should be ideally be only 1 file for this . So this is not expected.